File: nim/rna-transcription/rna_transcription.nim
import strutils

const Nucleotides = {'G', 'C', 'T', 'A'}

proc toRna*(dna: string): string =
  if not dna.allCharsInSet(Nucleotides):
    raise newException(ValueError, "Invalid nucleotide")

  dna.multiReplace(
    ("G", "C"),
    ("C", "G"),
    ("T", "A"),
    ("A", "U")
  )

File: fsharp/rna-transcription/RnaTranscription.fs
module RnaTranscription

let private dnaToRna = 
    function
    | 'G' -> 'C'
    | 'C' -> 'G'
    | 'T' -> 'A'
    | 'A' -> 'U'
    | _   -> invalidOp "Invalid nucleotide"

let toRna = String.map dnaToRna